lib/enigma.rb

```diff
@@ -1,8 +1,40 @@
+require 'date'
 class Enigma
+  def rand_key
+    rand(99999)
+  end
 
-  def pad(number)
-    pad_num = number.to_s
+  def zero_pad(key)
+    pad_num = key.to_s
     pad_num.insert(0, "0") until pad_num.size == 5
     pad_num
   end
+
+  def shift_keys(key)
+    keys = []
+    key.chars.each_with_index do |char, index|
+      next if index == 4
+      keys << char + key[index + 1]
+    end
+    keys
+  end
+
+  def date
+    date = Date.today.strftime("%d%m%y")
+  end
```

Comment:
"date" just grabs today's date and formats it into DDMMYY format, Then produces a string in that format. 
Also pretty short and in every cipher, should probably be an instance variable.